### DISCUSSION TITLE
How to load large models?

### DISCUSSION THREAD

**Discussion by aesher9o1** [kubeflow/kfserving#2352] [Q&A]
We have written a custom predictor for KServe. The model is loaded into the GPU directly and due to the size of the model it usually takes 3–4 mins to load. 

Here is how our inference service looks

    apiVersion: serving.kserve.io/v1beta1
    kind: InferenceService
    metadata:
      name: custom-model
    spec:
      predictor:
        containers:
          - name: kserve-container
            image: private-registry/kserve:1.0.3
            readinessProbe:
              httpGet:
                path: /
                port: 8080
              failureThreshold: 100
              initialDelaySeconds: 300
              periodSeconds: 300
            resources:
              requests:
                nvidia.com/gpu: 1
                cpu: 6000m
                memory: 16Gi
              limits:
                nvidia.com/gpu: 1
                cpu: 6000m
                memory: 16Gi

But the readiness checks is not taken into consideration by the queue-proxy container which starts calling the model container as soon as it starts. 

[![enter image description here][1]][1]

  [1]: https://i.stack.imgur.com/bcjGJ.png

Is there a way to pause the queue-proxy to wait for the model to load?

**Comment by yuzisun** [comment]
@aesher9o1 Those are actually probes waiting for the model container to be ready. The queue proxy is not marked ready until model server is ready.
